src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Startup.cs

```diff
@@ -29,19 +29,15 @@ public override void Configure(IFunctionsHostBuilder builder)
                 .AddDbContext<ContentDbContext>(options =>
                     options.UseSqlServer(ConnectionUtils.GetAzureSqlConnectionString("ContentDb"),
                         providerOptions => providerOptions.EnableCustomRetryOnFailure()))
-                .AddSingleton<IBlobStorageService, BlobStorageService>(
-                    provider =>
-                    {
-                        var connectionString = GetConfigurationValue(provider, "CoreStorage");
-
-                        var blobStorageService = new BlobStorageService(
-                            connectionString,
-                            new BlobServiceClient(connectionString),
-                            provider.GetRequiredService<ILogger<BlobStorageService>>(),
-                            new StorageInstanceCreationUtil()
-                        );
-                        return blobStorageService;
-                    })
+                .AddSingleton<IPrivateBlobStorageService, BlobStorageService>(provider =>
+                {
+                    var privateConnectionString = GetConfigurationValue(provider, "CoreStorage");
```

Comment:
I guess one of the advantages of having PublicBlobStorageService and PrivateBlobStorageService concrete classes that extend BlobStorageService is that you could have the class itself extract the connection string that relates to it rather than having to wire it manually here.  Then you could write here instead:

```
    .AddSingleton<IPrivateBlobStorageService, PrivateBlobStorageService>
```

and be done with it.  Within PrivateBlobStorageService, it would take the Configuration as a constructor argument and extract the "CoreStorage" string from it.

Reply:
And in this case it would also create its own `new BlobServiceClient(privateConnectionString)` within its constructor too.